### ISSUE TITLE
Framework should not inject `TKGVERSION` annotation into newly created management clusters

### ISSUE THREAD

**Issue by joshrosso** [vmware-tanzu/tanzu-framework#1259]
**Bug description**

Since TKG is only 1 distribution of framework, newly created management clusters should **not** be injecting a TKG version annotation on the Cluster object. Specifying management cluster version (per edition/distro and its version) is high-value. So this should be pluggable based on edition:version detection.

You'll find this logic present here: https://github.com/vmware-tanzu/tanzu-framework/blob/166919ff9c388fbc68f2717c7e6dd48811fcf853/pkg/v1/tkg/clusterclient/clusterclient.go#L968-L975

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Comment by danniel1205** [comment]
Thanks for filing this issue. TKG Telemetry currently reads the version from the `TKGVersionKey` of CAPI cluster object. We need to find an alternative for telemetry if we decide to take TKG version out.

**Comment by anujc25** [comment]
This value is heavily used to determine the version of TKG management clusters. Currently, **this value is determined from the TKG BoM file** and has been added as an annotation on the management cluster to denote the version of the management cluster. 

Currently, This value gets used in many places including upgrade validation, plugin version compatibility.

Sample TKG BoM file:
```
apiVersion: run.tanzu.vmware.com/v1alpha2
default:
  k8sVersion: v1.22.3+vmware.1-tkg.2-zshippable
release:
  version: v1.4.0 # This value is used to set TKGVersionKey as part of the annotation
```

> Specifying management cluster version (per edition/distro and its version) is high-value.

If we would like to do this, maybe one possible way to do this is to use a different BoM file for each edition and specify a different version for that edition?

**Comment by vuil** [comment]
Yeah the value of the TKGVERSION is specific to the framework (assuming there is no customization to what that version of framework understands to be the TKG BOM), and whose values carries some implications to telemetry, upgrade and version compatibility checks. None of those implications cannot be addressed, some hard than others, but if it is all the same, @joshrosso would introducing another annotation to set the EDITION work as well?

**Comment by joshrosso** [comment]
> TKGVERSION is specific to the framework

Seems like this should be specific to TKG, not framework, right?

Framework exists to bootstrap different distributions/editions/etc, of which examples are:

* Tanzu Kubernetes Grid (TKG, product)
* Tanzu Community Edition (TCE, OSS project)

So the crux of my point is more philosophical. We need to move away from binding TKG product details into framework.

So I'm not saying TKG should not add this annotation, I am saying framework should not do it by default.

> @joshrosso would introducing another annotation to set the EDITION work as well?

Yes, we can work with this to resolve the near term problem.

**Comment by vuil** [comment]
> > TKGVERSION is specific to the framework
> 
> Seems like this should be specific to TKG, not framework, right?
It is actually tied to the framework version. It's probably more a misnomer that is causing the confusion. The annotation key should perhaps be something like TANZUVERSION or just VERSION. The fact is that the version of framework (or the version of the plugin responsible for cluster LCM) determines the BOM used which in turn determines the value set for said annotation. And some cluster LCM logic relies on said value. So if a distribution/edition consumes that framework version and relies on it for cluster LCM (e.g. TCE using it to create management cluster), it is expected to result in the same value getting annotated on that cluster. 


> So the crux of my point is more philosophical. We need to move away from binding TKG product details into framework.
Agree, but this is more a consequence having to directly consume cluster LCM related functionality directly via framework. There are some thoughts of refactoring the repository. I would expect when that happens responsibilities like setting such annotations should no longer be within framework any more.

> > @joshrosso would introducing another annotation to set the EDITION work as well?
> 
> Yes, we can work with this to resolve the near term problem.
Cool. Unless there are additional suggestion we will look into providing an annotation for the edition for this issue.

**Comment by timothysc** [comment]
This may no longer be necessary with the new proposals on TKr coming in.  cc @vijaykatam

**Comment by joshrosso** [comment]
Largely the ask is to be able to make this pluggable so we can discern which channel or edition a management cluster was created with.

Feels like it'd be too heave of a lift to back out of the TKGVERSION field, so I'll propose a PR that can at-least add the bootstrapping distribution/edition.
